File: native-ios/MGLFleetSDK/Sources/MGLFleetSDK/FleetPresentationBridge.swift
import Foundation

enum FleetPresentationBridge {
    static var pendingSession: FleetSessionOptions?
    static var pendingCompletion: FleetSdkCompletionHandler?
    static var listeners: [(String, FleetSdkEventListener)] = []

    static func emit(name: String, payload: [String: Any]? = nil) {
        listeners.forEach { _, listener in
            listener.onEvent(name: name, payload: payload)
        }
    }
}

File: native-android/fleet-sdk/src/main/java/com/mgl/fleet/sdk/internal/FleetPresentationBridge.kt
package com.mgl.fleet.sdk.internal

import com.mgl.fleet.sdk.FleetSdkCompletionCallback
import com.mgl.fleet.sdk.FleetSdkEventListener
import com.mgl.fleet.sdk.FleetSessionOptions

internal object FleetPresentationBridge {
    var pendingSession: FleetSessionOptions? = null
    var pendingCallback: FleetSdkCompletionCallback? = null
    val listeners = mutableListOf<Pair<String, FleetSdkEventListener>>()

    fun emit(name: String, payload: Map<String, Any?>? = null) {
        listeners.forEach { (_, listener) ->
            listener.onEvent(name, payload)
        }
    }
}
